src/mlpack/methods/ann/layer/lp_pooling.hpp

```diff
@@ -0,0 +1,284 @@
+
+  //! Get the output size.
+  size_t OutputSize() const { return outSize; }
+
+  //! Get the normType.
+  size_t NormType() const { return normType; }
+  //! Modify the normType.
+  size_t& NormType() { return normType; }
+
+  //! Get the kernel width.
+  size_t KernelWidth() const { return kernelWidth; }
+  //! Modify the kernel width.
+  size_t& KernelWidth() { return kernelWidth; }
+
+  //! Get the kernel height.
+  size_t KernelHeight() const { return kernelHeight; }
+  //! Modify the kernel height.
+  size_t& KernelHeight() { return kernelHeight; }
+
+  //! Get the stride width.
+  size_t StrideWidth() const { return strideWidth; }
+  //! Modify the stride width.
+  size_t& StrideWidth() { return strideWidth; }
+
+  //! Get the stride height.
+  size_t StrideHeight() const { return strideHeight; }
+  //! Modify the stride height.
+  size_t& StrideHeight() { return strideHeight; }
+
+  //! Get the value of the rounding operation
+  bool const& Floor() const { return floor; }
+  //! Modify the value of the rounding operation
+  bool& Floor() { return floor; }
+
+  //! Get the size of the weights.
+  size_t WeightSize() const { return 0; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t /* version */);
+
+ private:
+  /**
+   * Apply pooling to the input and store the results.
+   *
+   * @param input The input to be apply the pooling rule.
+   * @param output The pooled result.
+   */
+  template<typename eT>
+  void Pooling(const arma::Mat<eT>& input, arma::Mat<eT>& output)
+  {
+    for (size_t j = 0, colidx = 0; j < output.n_cols;
+         ++j, colidx += strideHeight)
+    {
+      for (size_t i = 0, rowidx = 0; i < output.n_rows;
+           ++i, rowidx += strideWidth)
+      {
+        arma::mat subInput = input(
+            arma::span(rowidx, rowidx + kernelWidth - 1 - offset),
+            arma::span(colidx, colidx + kernelHeight - 1 - offset));
+
+        output(i, j) = pow(arma::accu(arma::pow(subInput,
+            normType)), 1.0/normType);
+      }
+    }
+  }
+
+  /**
+   * Apply unpooling to the input and store the results.
+   *
+   * @param input The input to be apply the unpooling rule.
+   * @param output The pooled result.
+   */
+  template<typename eT>
+  void Unpooling(const arma::Mat<eT>& input,
+                 const arma::Mat<eT>& error,
+                 arma::Mat<eT>& output)
+  {
+    const size_t rStep = input.n_rows / error.n_rows - offset;
+    const size_t cStep = input.n_cols / error.n_cols - offset;
+
+    arma::Mat<eT> unpooledError;
+    for (size_t j = 0; j < input.n_cols - cStep; j += cStep)
+    {
+      for (size_t i = 0; i < input.n_rows - rStep; i += rStep)
+      {
+        const arma::Mat<eT>& inputArea = input(arma::span(i, i + rStep - 1),
+            arma::span(j, j + cStep - 1));
+        size_t sum = pow(arma::accu(arma::pow(inputArea, normType)),
+            (normType-1) / normType);
```

Comment:
```suggestion
            (normType - 1) / normType);
```
Same as above. Thanks.